### PR TITLE
Fix icon shape color for colored icons.

### PR DESCRIPTION
**Pull request by saurabhg772244**
## :bookmark_tabs: Summary

- Styles icon shape so as not to override icons default colour.
- Fix jagged edges on icon shape.

Resolves #<your issue id here>

## :straight_ruler: Design Decisions

- All the icons that support will have fill as current colour. So will use css style `colour` to override it. All icons with fill as custom colour hexcode will not get overridden. 
- Used stroke transparent while drawing background of icon shape. this prevents jagged edges.

### :clipboard: Tasks

Make sure you

- [X] :book: have read the [contribution guidelines](https://mermaid.js.org/community/contributing.html)
- [X] :computer: have added necessary unit/e2e tests.
- [ ] :notebook: have added documentation. Make sure [`MERMAID_RELEASE_VERSION`](https://mermaid.js.org/community/contributing.html#update-documentation) is used for all new features.
- [x] :butterfly: If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### CODE REVIEW

File: packages/mermaid/src/docs/icon-shape.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+- **label**: The text label associated with the icon. This can be any string. If not defined, no label will be displayed.
+- **pos**: The position of the label. Possible values are:
+  - `t` (top)
+  - `b` (bottom, default if not provided)
+- **h**: The height of the icon in pixels. This parameter is optional and defaults to 48 pixels, which is the minimum height.
+
+### Example
+
+```mermaid
+B2@{ icon: "fa:window-minimize", form: "rounded", label: "B2", pos: "t", h: 60 }
```
[the remark this answers is not in the document]
Comment (aloisklink) [reply packages/mermaid/src/docs/icon-shape.md:31]: Also, ideally, we should use an icon that's available on the version of Mermaid that's on the https://mermaid.js.org/ website.

I'm guessing that might require modifying https://github.com/mermaid-js/mermaid/blob/0edfab104886de283ca6c4741ff1568f03670f77/packages/mermaid/src/docs/.vitepress/theme/Mermaid.vue#L87-L103 though, so it's not required for this PR.

**Review by aloisklink — CHANGES_REQUESTED** [review]
All your changes in the Mermaid code look good!

My recommended changes are:

- Can we change the test icon to something that's not a logo?
- Can you add a changeset using `pnpm changeset`? Maybe you can even make two, since you're fixing two different bugs in this PR! Please also update the PR title as well (even if it's just a general "Icon shapes improvements" since this PR handles a lot of stuff).
- The added documentation in `packages/mermaid/src/docs/icon-shape.md` needs a bit of work to look good on the https://mermaid.js.org/ website

**Comment by saurabhg772244** [comment]
@aloisklink I have updated the icon used in visual test. Also removed Doc from this PR. Will create separate PR for doc changes.

File: cypress/integration/rendering/iconShape.spec.ts [hunk 2]
```diff
@@ -1,3 +1,24 @@
...
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE */
```
Comment (aloisklink) [inline cypress/integration/rendering/iconShape.spec.ts:21]: :+1: Could we move this to right before the SVG is actually defined in https://github.com/mermaid-js/mermaid/blob/a98290a6e2e4fb4e382ae0f3a495c80a1d47639c/cypress/platform/viewer.js#L70 ?

**Review by aloisklink — CHANGES_REQUESTED** [review]
Moving the docs into a separate PR sounds good to me!

Two more changes required!

- Please move the license for `tropical-fish` to the correct file (e.g. where the SVG is defined)!
- And please use `pnpm changeset` to create a changeset for each of the bugs you've fixed!

File: packages/mermaid/src/docs/icon-shape.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (aloisklink) [reply packages/mermaid/src/docs/icon-shape.md]: Removed in https://github.com/mermaid-js/mermaid/pull/5937/commits/a98290a6e2e4fb4e382ae0f3a495c80a1d47639c

File: packages/mermaid/src/docs/icon-shape.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+```
+
+In this example:
+
+- The icon is "fa:window-minimize".
+- The shape is "rounded".
+- The label is "B2".
+- The label position is at the top (`t`).
+- The height of the icon is 40 pixels.
```
[the remark this answers is not in the document]
Comment (aloisklink) [reply packages/mermaid/src/docs/icon-shape.md]: Removed in https://github.com/mermaid-js/mermaid/pull/5937/commits/a98290a6e2e4fb4e382ae0f3a495c80a1d47639c

File: packages/mermaid/src/docs/icon-shape.md
[the remark this answers is not in the document]
Comment (aloisklink) [reply packages/mermaid/src/docs/icon-shape.md:1]: Removed in https://github.com/mermaid-js/mermaid/pull/5937/commits/a98290a6e2e4fb4e382ae0f3a495c80a1d47639c

File: packages/mermaid/src/docs/icon-shape.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  },
+]);
+```
+
+In this example:
+
+- The `logos` icon pack is registered by fetching the icons from the specified URL.
+- The `fa` (Font Awesome) icon pack is registered similarly.
+
+Ensure that the URLs provided point to valid Iconify JSON files containing the icon definitions.
```
[the remark this answers is not in the document]
Comment (aloisklink) [reply packages/mermaid/src/docs/icon-shape.md]: Removed in https://github.com/mermaid-js/mermaid/pull/5937/commits/a98290a6e2e4fb4e382ae0f3a495c80a1d47639c

File: cypress/platform/viewer.js [hunk 5]
```diff
@@ -61,11 +61,26 @@ const contentLoaded = async function () {
...
+          height: 153,
+        },
+      },
+      width: 256,
+      height: 256,
+    };
```
[the remark this answers is not in the document]
Comment (aloisklink) [reply cypress/platform/viewer.js]: Partially fixed by https://github.com/mermaid-js/mermaid/pull/5937/commits/9bac464442667389a2cf3c19e7e75cb0e788c1e1

See https://github.com/mermaid-js/mermaid/pull/5937#discussion_r1792927161 for further issues.

File: cypress/integration/rendering/iconShape.spec.ts (hunk 2 above)
Comment (saurabhg772244) [reply cypress/integration/rendering/iconShape.spec.ts]: Updated it.

Comment (aloisklink) [reply cypress/integration/rendering/iconShape.spec.ts]: Fixed by https://github.com/mermaid-js/mermaid/pull/5937/commits/a00161946496a1976f3b67d2d322d43e4bc5e1a5

**Review by sidharthv96 — APPROVED** [review]

**Review by aloisklink — APPROVED** [review]

**Pull request merged**